### ISSUE TITLE
Contact Forms: loaded in many admin screens where it shouldn't be loaded

### ISSUE THREAD

**Issue by eliorivero** [Automattic/jetpack#7596]
#### Steps to reproduce the issue
1. install and activate Query Monitor
2. go to Comments

#### What I expected
If there was something related to CF, it should only be the menu

#### What happened instead
There are multiple enqueues and an editor registration
```
admin_notices

Grunion_Editor_View::handle_editor_view_js()
wp-content/plugins/jetpack/modules/contact-form/grunion-editor-view.php:57
```

**Comment by georgestephanis** [comment]
Yes, and that action immediately returns if the current screen is one it doesn't care about.

https://github.com/Automattic/jetpack/blob/19b0eec0c298c6029995539ab02796ffa78bf776/modules/contact-form/grunion-editor-view.php#L57-L61

**Comment by eliorivero** [comment]
Proposal:
https://github.com/Automattic/jetpack/compare/update/contact-form-current-screen

**Comment by jeherve** [comment]
We now only hook on specific screens. This should be good to close.
